dandiapi/api/tests/test_dandiset.py

```diff
@@ -129,7 +129,11 @@ def test_dandiset_rest_create(api_client, user):
     assert dandiset.versions.count() == 1
     assert dandiset.most_recent_version.version == 'draft'
     assert dandiset.most_recent_version.metadata.name == name
-    assert dandiset.most_recent_version.metadata.metadata == metadata
+    assert dandiset.most_recent_version.metadata.metadata == {
+        **metadata,
+        'name': name,
+        'identifier': DANDISET_SCHEMA_ID_RE,
```

Comment:
```suggestion
        'identifier': response.data['identifier'],
```

Reply:
This won't match because the value in the metadata isn't the raw identifier, but adheres to [this](https://github.com/satra/dandi-cli/blob/2f4755f19f8c6f5272313c6d440306ec1955a5ef/dandi/models.py#L677-L683) schema change (of form `DANDI:XXXXXX`)